hal/halx86/apic/apic.c

```diff
@@ -24,6 +24,11 @@
 
 /* GLOBALS ********************************************************************/
 
+extern KIRQL HalpIrqlSynchLevel;
+
+#undef SYNCH_LEVEL
```

Comment:
What about doing that in hal.h? That way we'd prevent it from being used wrongly somewhere else in the future.